packages/react-pages/src/client/state.ts

```diff
@@ -139,12 +134,12 @@ else {
   useTheme = () => initialTheme
   usePagePaths = () => initialPagePaths
   usePageModule = (path) => {
-    const page = initialPages[path] || initialPages['/404']
+    const page = initialPages[path]
```

Comment:
Why remove the `/404` fallback?

Reply:
Otherwise it will never pass this `if` condition.
It should let the `useAppState` to goto the `404` state. So that the 404 state will be passed to the theme.

Reply:
That's the point, though. If a 404 page exists, there's no need for the theme to handle 404s manually.

Reply:
Theme need to handle 404 because it have to render the 404 view inside the app layout. Example: 
https://github.com/vitejs/vite-plugin-react-pages/blob/fe609fa92e77478de60190131223e6de9c6ea9f7/packages/theme-basic/src/index.tsx#L87